linera-views/src/backends/lru_caching.rs

```diff
@@ -78,49 +95,73 @@ enum CacheEntry {
 /// range deletions.
 struct LruPrefixCache {
     map: BTreeMap<Vec<u8>, CacheEntry>,
-    queue: LinkedHashMap<Vec<u8>, (), RandomState>,
-    max_cache_size: usize,
+    queue: LinkedHashMap<Vec<u8>, usize, RandomState>,
+    storage_cache_config: StorageCacheConfig,
+    total_size: usize,
     /// Whether we have exclusive R/W access to the keys under the root key of the store.
     has_exclusive_access: bool,
 }
 
 impl LruPrefixCache {
     /// Creates an `LruPrefixCache`.
-    pub fn new(max_cache_size: usize) -> Self {
+    pub fn new(storage_cache_config: StorageCacheConfig) -> Self {
         Self {
             map: BTreeMap::new(),
             queue: LinkedHashMap::new(),
-            max_cache_size,
+            storage_cache_config,
+            total_size: 0,
             has_exclusive_access: false,
         }
     }
 
+    /// Clear the entries of the queue within the specified constraints.
+    fn clear_queue(&mut self) {
```

Comment:
"clear_queue" is not the right concept. Perhaps "clear_some_entries", "evict_some_entries" or "make_room"

Reply:
I prefer `trim_cache` as terminology.